pkg/download/protocol_test.go

```diff
@@ -317,3 +318,31 @@ func (m *mockFetcher) Fetch(ctx context.Context, mod, ver string) (*storage.Vers
 		Zip:  ioutil.NopCloser(bytes.NewReader(bts)),
 	}, nil
 }
+
+func TestDownloadProtocolWhenFetchFails(t *testing.T) {
+	s, err := mem.NewStorage()
+	if err != nil {
+		t.Fatal(err)
+	}
+	fakeMod := testMod{"github.com/athens-artifacts/samplelib", "v1.0.0"}
+	bts := []byte(fakeMod.mod + "@" + fakeMod.ver)
+	err = s.Save(context.Background(), fakeMod.mod, fakeMod.ver, bts, ioutil.NopCloser(bytes.NewReader(bts)), bts)
+	if err != nil {
+		t.Fatal(err)
+	}
+	mp := &NotFoundFetcher{}
+	st := stash.New(mp, s)
+	dp := New(&Opts{s, st, nil})
+	ctx := context.Background()
+	_, err = dp.GoMod(ctx, fakeMod.mod, fakeMod.ver)
+	if err != nil {
```

Comment:
If the `NotFoundFetcher` always returns an error, won't this error **always** not equal nil?